crates/bevy_pbr/src/render/shadow_sampling.wgsl

```diff
@@ -0,0 +1,133 @@
+}
+
+fn sample_shadow_map_jimenez_fourteen(light_local: vec2<f32>, depth: f32, array_index: i32, texel_size: f32) -> f32 {
+    let shadow_map_size = vec2<f32>(textureDimensions(view_bindings::directional_shadow_textures));
+
+    let random_angle = 2.0 * PI * interleaved_gradient_noise(light_local * shadow_map_size);
+    let m = vec2(sin(random_angle), cos(random_angle));
+    let rotation_matrix = mat2x2(
+        m.y, -m.x,
+        m.x, m.y
+    );
+
+    let f = map(0.00390625, 0.022949219, 0.015, 0.035, texel_size);
+    let uv_offset_scale = f / (texel_size * shadow_map_size);
+
+    // https://www.iryoku.com/next-generation-post-processing-in-call-of-duty-advanced-warfare (slides 120-135)
+    let sample_offset1 = (rotation_matrix * vec2(-0.7071,  0.7071)) * uv_offset_scale;
+    let sample_offset2 = (rotation_matrix * vec2(-0.0000, -0.8750)) * uv_offset_scale;
+    let sample_offset3 = (rotation_matrix * vec2( 0.5303,  0.5303)) * uv_offset_scale;
+    let sample_offset4 = (rotation_matrix * vec2(-0.6250, -0.0000)) * uv_offset_scale;
+    let sample_offset5 = (rotation_matrix * vec2( 0.3536, -0.3536)) * uv_offset_scale;
+    let sample_offset6 = (rotation_matrix * vec2(-0.0000,  0.3750)) * uv_offset_scale;
+    let sample_offset7 = (rotation_matrix * vec2(-0.1768, -0.1768)) * uv_offset_scale;
+    let sample_offset8 = (rotation_matrix * vec2( 0.1250,  0.0000)) * uv_offset_scale;
+
+    var sum = 0.0;
+    sum += sample_shadow_map_hardware(light_local + sample_offset1, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset2, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset3, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset4, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset5, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset6, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset7, depth, array_index);
+    sum += sample_shadow_map_hardware(light_local + sample_offset8, depth, array_index);
+    return sum / 8.0;
+}
+
+fn sample_shadow_map(light_local: vec2<f32>, depth: f32, array_index: i32, texel_size: f32) -> f32 {
+#ifdef SHADOW_FILTER_METHOD_CASTANO_13
+    return sample_shadow_map_castano_thirteen(light_local, depth, array_index);
+#else ifdef SHADOW_FILTER_METHOD_JIMENEZ_14
+    return sample_shadow_map_jimenez_fourteen(light_local, depth, array_index, texel_size);
+#else ifdef SHADOW_FILTER_METHOD_HARDWARE_2X2
+    return sample_shadow_map_hardware(light_local, depth, array_index);
+#else
+    // This needs a default return value to avoid shader compilation errors if it's compiled with no SHADOW_FILTER_METHOD_* defined.
+    // (eg. if the normal prepass is enabled it ends up compiling this due to the normal prepass depending on pbr_functions, which depends on shadows)
```

Comment:
No this is a new comment/issue we solved a week or so ago.

Reply:
could we rearrange it so if HARDWARE2x2 and JIMENEZ14 are not defined, we use the default method?

Reply:
I agree that this should be rearranged so that the default method (Castano13) is the one that is used if no shader defs are set.

Reply:
No, see the comment. This should never actually be used unless a SHADOW_FILTER shaderdef is already defined.
```
// This needs a default return value to avoid shader compilation errors if it's compiled with no SHADOW_FILTER_METHOD_* defined.
// (eg. if the normal prepass is enabled it ends up compiling this due to the normal prepass depending on pbr_functions, which depends on shadows)
// This should never actually get used, as anyone using bevy's lighting/shadows should always have a SHADOW_FILTER_METHOD defined.
// Set to 0 to make it obvious that something is wrong.
```

Reply:
The point of having it default to 0 instead of Castano was to basically make the error obvious in case something broke or someone forgot to pass in a `SHADOW_FILTER_METHOD` in future bevy updates or for custom pipelines.

Reply:
I'll accept it.